### PR TITLE
Migrate image tag tests from integration-cli to api tests

### PR DESCRIPTION
**Pull request by yongtang**
This fix migrates image tag tests from integration-cli to api tests.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: integration/image/tag_test.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+	insp, _, err := client.ImageInspectWithRaw(ctx, "busybox")
+	assert.NilError(t, err)
+
+	err = client.ImageTag(ctx, insp.ID, "testfoobarbaz")
+	assert.NilError(t, err)
+}
+
+// ensure we don't allow the use of invalid repository names; these tag operations should fail
+func TestTagInvalidUnprefixedRepo(t *testing.T) {
```
Comment (vdemeester) [inline integration/image/tag_test.go]: I think thoses should be unit tests instead.

File: integration/image/tag_test.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	err := client.ImageTag(ctx, "busybox:latest", "testfoobarbaz")
+	assert.NilError(t, err)
+}
+
+// tagging an image by ID in a new unprefixed repo should work
+func TestTagUnprefixedRepoByID(t *testing.T) {
```
Comment (vdemeester) [inline integration/image/tag_test.go]: Could we merge `TestTagUnprefixedRepoByName` and `TestTagUnprefixedRepoByID` in one tests ?

File: integration/image/tag_test.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	invalidRepos := []string{"fo$z$", "Foo@3cc", "Foo$3", "Foo*3", "Fo^3", "Foo!3", "F)xcz(", "fo%asd", "FOO/bar"}
+
+	for _, repo := range invalidRepos {
+		err := client.ImageTag(ctx, "busybox", repo)
+		testutil.ErrorContains(t, err, "not a valid repository/tag")
+	}
+}
+
+// ensure we don't allow the use of invalid tags; these tag operations should fail
+func TestTagInvalidPrefixedRepo(t *testing.T) {
```
Comment (vdemeester) [inline integration/image/tag_test.go]: Same as `TestTagInvalidUnprefixedRepo` — or at least we could merge both and name the test ` TestTagInvalidReference`.

File: integration/image/tag_test.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+func TestTagExistedNameWithoutForce(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	err := client.ImageTag(ctx, "busybox:latest", "busybox:test")
+	assert.NilError(t, err)
+}
+
+func TestTagWithPrefixHyphen(t *testing.T) {
```
Comment (vdemeester) [inline integration/image/tag_test.go]: Same as `TestTagInvalidUnprefixedRepo` and `TestTagInvalidReference`

File: integration/image/tag_test.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+	digest := "busybox@sha256:abcdef76720241213f5303bda7704ec4c2ef75613173910a56fb1b6e20251507"
+	// test setting tag fails
+	err := client.ImageTag(ctx, "busybox:latest", digest)
+	testutil.ErrorContains(t, err, "refusing to create a tag with a digest reference")
+	// check that no new image matches the digest
+	_, _, err = client.ImageInspectWithRaw(ctx, digest)
+	testutil.ErrorContains(t, err, fmt.Sprintf("No such image: %s", digest))
+}
+
+func TestTagInvalidRepoName(t *testing.T) {
```
Comment (vdemeester) [inline integration/image/tag_test.go]: Could be merged with `TestTagInvalidUnprefixedRepo` and `TestTagInvalidPrefixedRepo`

**Review by vdemeester — APPROVED** [review]
LGTM :tiger: with some comments

File: integration/image/tag_test.go (hunk 5 above)
Comment (yongtang) [reply integration/image/tag_test.go]: Thanks @vdemeester. Update.

File: integration/image/tag_test.go (hunk 4 above)
Comment (yongtang) [reply integration/image/tag_test.go]: Done.

File: integration/image/tag_test.go (hunk 3 above)
Comment (yongtang) [reply integration/image/tag_test.go]: Done.

File: integration/image/tag_test.go (hunk 2 above)
Comment (yongtang) [reply integration/image/tag_test.go]: Done.

File: integration/image/tag_test.go (hunk 1 above)
Comment (yongtang) [reply integration/image/tag_test.go]: Thanks @vdemeester. I added a `TODO` in the comment, and will work on it in a follow up PR.

**Comment by anusha-ragunathan** [comment]
LGTM

**Pull request merged**